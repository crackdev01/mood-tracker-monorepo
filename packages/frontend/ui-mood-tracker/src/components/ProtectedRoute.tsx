import React from 'react';
import { Route, Redirect } from 'react-router-dom';

const ProtectedRoute = ({ component, isAuthenticated, ...rest }: any) => {
  const routeComponent = (props: any) =>
    isAuthenticated ? (
      React.createElement(component, props)
    ) : (
      <Redirect to={{ pathname: '/login' }} />
    );
  return <Route {...rest} render={routeComponent} />;
};

export default ProtectedRoute;
