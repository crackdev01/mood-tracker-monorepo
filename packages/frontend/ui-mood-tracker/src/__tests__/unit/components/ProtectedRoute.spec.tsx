import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProtectedRoute from '../../../components/ProtectedRoute';

Enzyme.configure({ adapter: new Adapter() });

const mockComponent = () => {
  return <div>Mock Component</div>;
};

describe('ProtectedRoute', () => {
  // TODO: Can't differentiate with generated snapshots. Check for other possibilities.
  test('renders for unauthenticated Route', () => {
    const props = {
      component: mockComponent,
      isAuthenticated: false,
    };
    const w = shallow(<ProtectedRoute {...props} />);
    expect(w.debug()).toMatchSnapshot();
  });

  test('renders for authenticated Route', () => {
    const props = {
      component: mockComponent,
      isAuthenticated: true,
    };
    const w = shallow(<ProtectedRoute {...props} />);
    expect(w.debug()).toMatchSnapshot();
  });
});
