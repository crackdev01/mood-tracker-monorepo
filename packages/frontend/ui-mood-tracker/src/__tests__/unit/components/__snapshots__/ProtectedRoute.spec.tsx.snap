// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute renders for authenticated Route 1`] = `"<Route render={[Function: routeComponent]} />"`;

exports[`ProtectedRoute renders for unauthenticated Route 1`] = `"<Route render={[Function: routeComponent]} />"`;
